### DISCUSSION TITLE
Feature request: expand pylance to support IPython syntax (magics, etc).

### DISCUSSION THREAD

**Discussion by ryan-feeley** [microsoft/pylance-release#1868] [Ideas] [answered]
An IPython magic statement like `%load_ext autoreload` or `%timeit f()` is not valid pure python code, but it is valid IPython syntax, which one can use in the IPython shell or Jupyter notebook. These are popular development tools.

I believe would be useful enhancement for pylance to avoid giving errors on such IPython line and cell magics if the underlying file type in the editor is `.ipy` (standard extension for an IPython script file) or `*.ipynb` (jupyter notebook).

What is the current recommended workaround for such cases? Does one add a trailing `# type: ignore` to each line? Or does one break out of the zen of ipython and do something like 
```py
from IPython.core.getipython import get_ipython
get_ipython().run_line_magic('reload_ext', 'autoreload')
```

Is there any plan on the roadmap for pylance to be extended to IPython syntax?

**Comment by jakebailey** [comment, marked as answer]
I'm going to transfer this to an issue; we don't use discussions for feature requests or bug reports.

**Reply by jakebailey** [reply]
https://github.com/microsoft/pylance-release/issues/1879
